src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,421 @@
+                for ( final Allele originalAllele : originalGT ) {
+                    best.add(allelesToUse.contains(originalAllele) ? originalAllele : ref);
+                }
+                gb.noGQ();
+                gb.noPL();
+                gb.alleles(best);
+                break;
+        }
+    }
+
+    /**
+     * From a given variant context, extract a given subset of alleles, and update genotype context accordingly,
+     * including updating the PL's, and assign genotypes accordingly
+     * @param vc                                variant context with alleles and genotype likelihoods
+     * @param defaultPloidy                     ploidy to assume in case that {@code vc} does not contain that information
+     *                                          for a sample.
+     * @param allelesToKeep                      alleles to subset
+     * @param method                            whether to assign hard genotypes or leave as no-call
+     * @return                                  GenotypesContext with new PLs
+     */
+    private static GenotypesContext subsetGeneralPloidyAlleles(final VariantContext vc, final int defaultPloidy,
+                                                               final List<Allele> allelesToKeep, final GenotypeAssignmentMethod method) {
+        Utils.nonNull(vc, "vc is null");
+        Utils.nonNull(allelesToKeep, "allelesToKeep is null");
+        if (allelesToKeep.size() == 1) {
+            return GATKVariantContextUtils.subsetToRefOnly(vc, defaultPloidy);
+        }
+        final GenotypesContext newGTs = GenotypesContext.create();
+
+        // create the new genotypes
+        for ( final Genotype g : vc.getGenotypes() ) {
+            final int ploidy = g.getPloidy() > 0 ? g.getPloidy() : defaultPloidy;
+            if ( !g.hasLikelihoods() ) {
+                newGTs.add(GenotypeBuilder.create(g.getSampleName(), GATKVariantContextUtils.noCallAlleles(ploidy)));
+                continue;
+            }
+
+            final double[] originalLikelihoods = g.getLikelihoods().getAsVector();
+            final double[] newLikelihoods = subsettedGenotypeLikelihoods(originalLikelihoods, ploidy, vc.getAlleles(), allelesToKeep);
+
+            if (!GATKVariantContextUtils.isInformative(newLikelihoods) ) {
+                newGTs.add(GenotypeBuilder.create(g.getSampleName(), GATKVariantContextUtils.noCallAlleles(ploidy)));
+                continue;
+            }
+
+            final GenotypeBuilder gb = new GenotypeBuilder(g);
+            gb.PL(newLikelihoods);
+
+
+            // if we weren't asked to assign a genotype, then just no-call the sample
+            if (method == GenotypeAssignmentMethod.DO_NOT_ASSIGN_GENOTYPES || method == GenotypeAssignmentMethod.SET_TO_NO_CALL) {
+                gb.alleles(GATKVariantContextUtils.noCallAlleles(ploidy));
+            } else {
+                assignGenotypes(gb, newLikelihoods, allelesToKeep, ploidy);
```

Comment:
And existing behavior is still preserved because the different values of `GenotypeAssignmentMethod` were never actually used in code.